examples/basic/layouts/dashboard.py

```diff
@@ -33,59 +30,55 @@ def slider():
     source = ColumnDataSource(data=dict(x=x, y=y))
 
-    plot = figure(
-        y_range=(-10, 10), tools='', toolbar_location=None,
-        title="Sliders example")
+    plot = figure(y_range=(-10, 10), tools='', toolbar_location=None,
+                  title="Sliders example")
     plot.line('x', 'y', source=source, line_width=3, line_alpha=0.6)
 
-    amp_slider = Slider(start=0.1, end=10, value=1, step=.1, title="Amplitude")
-    freq_slider = Slider(start=0.1, end=10, value=1, step=.1, title="Frequency")
-    phase_slider = Slider(start=0, end=6.4, value=0, step=.1, title="Phase")
-    offset_slider = Slider(start=-5, end=5, value=0, step=.1, title="Offset")
+    amp = Slider(start=0.1, end=10, value=1, step=.1, title="Amplitude")
+    freq = Slider(start=0.1, end=10, value=1, step=.1, title="Frequency")
+    phase = Slider(start=0, end=6.4, value=0, step=.1, title="Phase")
+    offset = Slider(start=-5, end=5, value=0, step=.1, title="Offset")
 
-    callback = CustomJS(args=dict(source=source, amp=amp_slider, freq=freq_slider, phase=phase_slider, offset=offset_slider),
+    callback = CustomJS(args=dict(source=source, amp=amp, freq=freq, phase=phase, offset=offset),
                         code="""
-        const data = source.data;
         const A = amp.value;
         const k = freq.value;
         const phi = phase.value;
         const B = offset.value;
-        const x = data['x']
-        const y = data['y']
-        for (let i = 0; i < x.length; i++) {
-            y[i] = B + A*Math.sin(k*x[i]+phi);
-        }
-        source.change.emit();
+
+        const x = source.data.x
+        const y = Array.from(x, (x) => B + A*Math.sin(k*x+phi))
+        source.data = { x: x, y: y }
     """)
 
-    amp_slider.js_on_change('value', callback)
-    freq_slider.js_on_change('value', callback)
-    phase_slider.js_on_change('value', callback)
-    offset_slider.js_on_change('value', callback)
+    amp.js_on_change('value', callback)
+    freq.js_on_change('value', callback)
+    phase.js_on_change('value', callback)
+    offset.js_on_change('value', callback)
+
+    widgets = column(amp, freq, phase, offset, sizing_mode="stretch_width")
 
-    widgets = column(amp_slider, freq_slider, phase_slider, offset_slider)
     return [widgets, plot]
 
 
 def linked_panning():
-    N = 100
-    x = np.linspace(0, 4 * np.pi, N)
+    x = np.linspace(0, 4 * np.pi, 100)
     y1 = np.sin(x)
     y2 = np.cos(x)
     y3 = np.sin(x) + np.cos(x)
 
     s1 = figure(tools='pan')
     s1.circle(x, y1, color="navy", size=8, alpha=0.5)
+
     s2 = figure(tools='pan', x_range=s1.x_range, y_range=s1.y_range)
     s2.circle(x, y2, color="firebrick", size=8, alpha=0.5)
+
     s3 = figure(tools='pan, box_select', x_range=s1.x_range)
     s3.circle(x, y3, color="olive", size=8, alpha=0.5)
+
     return [s1, s2, s3]
 
-l = grid([
-    bollinger(),
-    slider(),
-    linked_panning(),
-], sizing_mode='stretch_both')
+l = grid([bollinger(), slider(), linked_panning()],
```

Comment:
FYI @mattpap for me the center row looks pretty poor with lots of blank space in the center. I tried some sizing mode changes but nothing seemed to help.